### PR TITLE
Add logging to manifest object tagging (#3982)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Connected issue: #3982

### Author

- [x] Target branch is `develop`
- [x] Source branch is `issues/<github-handle>/<issue#>-<slug>`
- [x] PR title references the connected issue
- [x] PR title matches<sup>1</sup> title of connected issue         <sub>or comment in PR explains why they're different</sub>
- [x] Title of at least one commit references connected issue
- [x] PR is connected to issue via Zenhub 
- [x] PR description links to connected issue

<sup>1</sup> when the issue title describes a problem, the PR title is `Fix: ` followed by the issue title   

### Author (reindex)

- [x] Added `r` tag to commit title                                 <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                                   <sub>or this PR does not require reindexing</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain                <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                        <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst          <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                                 <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                                   <sub>or this PR does not require upgrading</sub>

### Author (operator tasks)

- [x] Added checklist items for additional operator tasks           <sub>or this PR does not require additional tasks</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title                            <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfix connected to the issue          <sub>or there is no temporary hotfix for the connected issue on the `prod` branch</sub>

### Author (requirements)

- [x] Ran `make requirements_update`                                <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                                 <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                                      <sub>or this PR does not touch requirements*.txt</sub>

### Author (rebasing, integration test)

- [x] `make integration_test` passes in personal deployment         <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after rejection)

Uncheck the *Author (requirements)* and *Author (rebasing, integration test)* 
checklists.

### Primary reviewer (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issue as `demo` or `no demo`
- [x] Commented on connected issue about demo expectations          <sub>or labelled connected issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Ticket is in *Review requested* column
- [x] Requested review from peer reviewer
- [x] Assigned PR to peer reviewer

### Peer reviewer (after rejection)

Uncheck the *Author (requirements)* and *Author (rebasing, integration test)* 
checklists.

### Peer reviewer (after approval)

- [x] `N reviews` label on PR reflects prior peer reviews with changes requested
- [x] Moved ticket to *Approved* column
- [x] Assigned PR to current operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear                      <sub>or connected issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Branch pushed to GitLab and added `sandbox` label             <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                                       <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                                  <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                             <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issue to Merged column
- [ ] Pushed merge commit to GitHub

### Operator (after pushing the merge commit)

- [ ] Shortened the PR chain                                        <sub>or this PR is not labeled `chain`</sub>
- [ ] Pushed merge commit to GitLab                                 <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from GitHub and GitLab
- [ ] Build passes on GitLab

### Operator (reindex) 

- [ ] Started reindex in ``dev``                                    <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures                     <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment                      <sub>or this PR does not require reindexing</sub>

### Operator

- [ ] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Comment by danielsotirhos** [comment]
Example:

Field | Value
-- | --
@ingestionTime | 1648510342870
@log | 122796619775:/aws/lambda/azul-service-daniel-manifest
@logStream | 2022/03/28/[$LATEST]250af52a04ab4b23bacb1f55edea3744
@message | [INFO] 2022-03-28T23:32:19.036Z fd5b0c7e-6b97-4ffb-86ce-4bab9585d4c1 Putting tag {'TagSet': [{'Key': 'azul_file_name', 'Value': 'HumanMousePancreas 2022-03-28 23.32.tsv'}]} on object manifests/bf34f1d3-2881-5321-a996-a0099788c873.5c957295-df8a-548b-9dd0-f75082d9816d.tsv
@requestId | fd5b0c7e-6b97-4ffb-86ce-4bab9585d4c1
@timestamp | 1648510339036
TagSet.0.Key | azul_file_name
TagSet.0.Value | HumanMousePancreas 2022-03-28 23.32.tsv

File: src/azul/service/storage_service.py [hunk 1]
```diff
@@ -206,6 +206,7 @@ def create_bucket(self, bucket_name: Optional[str] = None):
     def put_object_tagging(self, object_key: str, tagging: Tagging = None):
         deadline = time.time() + 60
         tagging = {'TagSet': [{'Key': k, 'Value': v} for k, v in tagging.items()]}
+        logger.info('Putting tag %s on object %s', tagging, object_key)
```
Comment (hannes-ucsc) [inline src/azul/service/storage_service.py:209]:
```suggestion
        logger.info('Tagging object %r with %r', object_key, tagging)
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#18string-interpolation

**Comment by danielsotirhos** [comment]
Example:

Field | Value
-- | --
@ingestionTime | 1648743661731
@log | 122796619775:/aws/lambda/azul-service-daniel-manifest
@logStream | 2022/03/31/[$LATEST]a9c764272f1e482fa135a4ebab1c9dee
@message | [INFO] 2022-03-31T16:20:58.467Z 06fbbc27-5391-422a-b628-4d2ad91cff40 Tagging object 'manifests/ef767aa2-5e05-5c32-ba9e-767d16973e88.5c957295-df8a-548b-9dd0-f75082d9816d.tsv' with {'TagSet': [{'Key': 'azul_file_name', 'Value': 'AcuteSkinInflammation 2022-03-31 16.20.tsv'}]}
@requestId | 06fbbc27-5391-422a-b628-4d2ad91cff40
@timestamp | 1648743658468
TagSet.0.Key | azul_file_name
TagSet.0.Value | AcuteSkinInflammation 2022-03-31 16.20.tsv

**Review by hannes-ucsc — APPROVED** [review]

**Review by noah-aviel-dove — APPROVED** [review]

**Pull request merged**